discovery-provider/src/queries/queries.py

```diff
@@ -196,7 +194,19 @@ def get_tracks_including_unlisted():
         current_user_id = get_current_user_id(required=False)
         extended_tracks = populate_track_metadata(session, track_ids, tracks, current_user_id)
 
-    return api_helpers.success_response(extended_tracks)
+        filtered_extended_tracks = []
+
+        def check_unlisted_track_for_proper_route_id(track):
+            if track["is_unlisted"] and track["route_id"] != helpers.create_track_route_id(i["url_title"], i["handle"]):
+                return False
+            
+            return True
```

Comment:
We could just return `track["is_unlisted"] and track["route_id"] != helpers.create_track_route_id(i["url_title"], i["handle"])` - no need for the if/else.

In general whenever you see this pattern: 
```
if (condition) { 
   return true
 } else {
   return false
}
```

It can be refactored into `return condition`